lib/Basics/process-utils.cpp

```diff
@@ -1565,3 +1567,29 @@ void TRI_ShutdownProcess() {
   }
   ExternalProcesses.clear();
 }
+
+std::string TRI_SetPriority(ExternalId pid, int prio) {
+#ifndef _WIN32
+  errno = 0;
+  int ret = setpriority(PRIO_PROCESS, pid._pid, prio);
+  if (ret == -1) {
+    int err = errno;
+    if (err != 0) {
+      std::stringstream ss;
+      ss << "setting process priority for : '" << pid._pid
+         << "' failed with error: " << strerror(err);
+      return ss.str();
+    }
+  }
+#else
+  HANDLE processHandle = OpenProcess(PROCESS_SUSPEND_RESUME, FALSE, pid._pid);
+  DWORD dwPriorityClass;
+  if (prio == 0) {
+    dwPriorityClass = NORMAL_PRIORITY_CLASS;
+  } else {
+    dwPriorityClass = IDLE_PRIORITY_CLASS;
+  }
+  SetPriorityClass(hCurrentProcess, dwPriorityClass);
```

Comment:
In order to use SetPriorityClass you have to open the handle with `PROCESS_SET_INFORMATION` instead of `PROCESS_SUSPEND_RESUME`.